BUILD

```diff
@@ -0,0 +1,133 @@
+load("@rules_foreign_cc//foreign_cc:defs.bzl", "cmake", "configure_make")
+
+cc_library(
+    name = "asio",
+    hdrs = ["@asio//:all_hdrs"],
+    strip_include_prefix = "include",
+    visibility = ["//visibility:public"]
+)
```

Comment:
for all dependencies expect foonathan_memory, it is possible to use a git submodule via a cmake option (see [here](https://fast-dds.docs.eprosima.com/en/latest/installation/configuration/cmake_options.html#third-party-libraries-options)). I think that'd make sense

Reply:
I ran into issues running this in a cloud build environment (github action). Thats why i opted for independent builds. Not sure if the problem persists outside gh actions.